File: E-Tour/src/main/java/com/etour/pricing/FareBand.java
package com.etour.pricing;

public enum FareBand {
    TWIN_SHARING("Twin sharing (per person)"),
    SINGLE("Single occupancy"),
    EXTRA_PERSON("Extra person in room"),
    CHILD_WITH_BED("Child with bed"),
    CHILD_WITHOUT_BED("Child without bed");

    private final String label;

    FareBand(String label) {
        this.label = label;
    }

    public String getLabel() {
        return label;
    }

    public static FareBand fromOccupancy(String value) {
        if (value == null || value.isBlank()) {
            return null;
        }
        String v = value.trim().toUpperCase();
        if (v.equals("ADULT")) {
            return TWIN_SHARING;
        }
        if (v.equals("SINGLE_PERSON")) {
            return SINGLE;
        }
        for (FareBand b : values()) {
            if (b.name().equals(v)) {
                return b;
            }
        }
        return null;
    }
}

File: ETour-DotNet/Pricing/FareBand.cs
namespace ETour.Api.Pricing;

public enum FareBand
{
    TWIN_SHARING,
    SINGLE,
    EXTRA_PERSON,
    CHILD_WITH_BED,
    CHILD_WITHOUT_BED
}

public static class FareBandExtensions
{
    public static string Label(this FareBand band) => band switch
    {
        FareBand.TWIN_SHARING => "Twin sharing (per person)",
        FareBand.SINGLE => "Single occupancy",
        FareBand.EXTRA_PERSON => "Extra person in room",
        FareBand.CHILD_WITH_BED => "Child with bed",
        FareBand.CHILD_WITHOUT_BED => "Child without bed",
        _ => band.ToString()
    };

    public static FareBand? FromOccupancy(string value)
    {
        if (string.IsNullOrWhiteSpace(value)) return null;

        var v = value.Trim().ToUpperInvariant();
        if (v == "ADULT") return FareBand.TWIN_SHARING;
        if (v == "SINGLE_PERSON") return FareBand.SINGLE;

        return Enum.TryParse<FareBand>(v, out var parsed) ? parsed : null;
    }
}
